website/docs/schema-generator/customizing-schemas/deprecating-schema.md

```diff
@@ -25,7 +33,22 @@ type Query {
 }
 ```
 
-While you can deprecate any fields/functions/classes in your Kotlin code, GraphQL only supports deprecation directive on
-the fields (which correspond to Kotlin properties and functions), input fields and enum values.
+## GraphQLDeprecated
 
-Deprecation of arguments is currently not supported [in Kotlin](https://youtrack.jetbrains.com/issue/KT-25643).
+A side-effect of using `@Deprecated` is that it marks your own code as being deprecated, which may not be what you want.
+
+![](https://user-images.githubusercontent.com/459464/182555658-49c5252a-b421-437a-b7a7-93c17d778a09.png)
+
+Find yourself using `Suppress("DEPRECATION")` everywhere?
+
+Then you may want to use `@GraphQLDeprecated` instead. It works just the same:
+
+
+```kotlin
+class SimpleQuery {
+  @GraphQLDeprecated(message = "this query is deprecated", replaceWith = ReplaceWith("shinyNewQuery"))
+  fun simpleDeprecatedQuery(): Boolean = false
+
+  fun shinyNewQuery(): Boolean = true
+}
+```
```

Comment:
```suggestion
A side-effect of using `@Deprecated` is that it marks your own Kotlin code as being deprecated, which may not be what you want. Using `@GraphQLDeprecated` you can add the `@deprecated` directive to the GraphQL schema, but not have your Kotlin code show up as deprecated in your editor.

```kotlin
class SimpleQuery {
  @GraphQLDeprecated(message = "this query is deprecated", replaceWith = ReplaceWith("shinyNewQuery"))
  fun simpleDeprecatedQuery(): Boolean = false

  fun shinyNewQuery(): Boolean = true
}
```
```